### PR TITLE
feat(astro): upgrade to Astro 3.0

### PR DESCRIPTION
**Pull request by floppydisken**
Upgrade to Astro 3.0 with all the new goodies.

BREAKING CHANGE: astro package now requires typescript 5

## What it does
Upgrade the nxtensions package to the Astro 3.0 release and adds the now external package @astrojs/check [Astro 3.0 breaking changes](https://docs.astro.build/en/guides/upgrade-to/v3/) for the check executor

## Related Issue(s)
#488 

Fixes #
Support for Astro 3.0

### CODE REVIEW

File: packages/astro/package.json [hunk 1]
```diff
@@ -33,9 +33,9 @@
     "@nx/js": "^16.0.0",
     "@nx/linter": "^16.0.0",
     "@nx/workspace": "^16.0.0",
-    "astro": "^1.6.0 || ^2.0.0",
+    "astro": "^1.6.0 || ^2.0.0 || ^3.0.0",
```
[the remark this answers is not in the document]
Comment (floppydisken) [reply packages/astro/package.json]: Done

**Comment by louiss0** [comment]
If you know how to add an executor can you do one for `astro add` Please we all need that one.

**Review by louiss0 — APPROVED** [review]
I think these changes are good. I should be able to update my project after these changes.

**Comment by louiss0** [comment]
I hope he can also add an executor for `astro add` #496 Maybe you can do so if not please support me

**Comment by CreaDisc** [comment]
Feels out-of-scope for this PR and Integrations can still be setup manually. Can we merge this first so people can have their Astro 3.0 support?

**Review by CreaDisc — APPROVED** [review]

**Comment by floppydisken** [comment]
@louiss0 I wouldn't mind implementing it. I'll do it in another branch after this has been merged so the people can get access to Astro 3.0

**Comment by louiss0** [comment]
> @louiss0 I wouldn't mind implementing it. I'll do it in another branch after this has been merged so the people can get access to Astro 3.0

Good

**Comment by NachoVazquez** [comment]
Would it be out of scope to upgrade the Nx versions?

**Comment by floppydisken** [comment]
@NachoVazquez probably not, since this would be a major release anyways, might as well upgrade the nx packages.

**Comment by Bjeaurn** [comment]
Hoping this gets merged and released soon! :) Nice job.

**Comment by KelechiOdom10** [comment]
Any updates on this? I hope this gets released soon! Thanks for the good work 😄

**Comment by mimarec** [comment]
The extension has been tested in an NX workspace? I am generating the buil locally from this pull request and consuming it in an NX workspace and it does not work.

**Comment by mimarec** [comment]
This pull request doesn't work locally because it hasn't taken into account the change in Astro's default export, which causes the executors to malfunction.

I have addressed the issue in this pull request:

https://github.com/nxtensions/nxtensions/pull/512

**Comment by jjtudela** [comment]
> This pull request doesn't work locally because it hasn't taken into account the change in Astro's default export, which causes the executors to malfunction.
> 
> I have addressed the issue in this pull request:
> 
> #512

Exactly, it doesn't work locally

**Comment by floppydisken** [comment]
Closing this in favor of #512 by @mimarec.

**Pull request closed without merging**